R/fileupload.R

```diff
@@ -20,6 +20,32 @@
 # form upload, i.e. traditional HTTP POST-based file upload) doesn't work with
 # the websockets package's HTTP server at the moment.
 
+#' @details \code{sanitizeFileName} is an effort to safely retain something
+#'   close to an uploaded file's original name in order to support libraries
+#'   like readxl that are sensitive to the names of files. A good overview of
+#'   the risks associated with uploaded files is
+#'   \url{https://www.owasp.org/index.php/Unrestricted_File_Upload}
+#' @param name A string, the name of a file to sanitize.
+#' @param defaultName The string to return if \code{name} cannot be sanitized.
+#' @param maxSize The maximum allowable size of \code{name} after sanitization.
+#'   (Default is 255 characters.)
+#' @return If \code{name} is an empty string, returns \code{defaultName}.
+#'   Otherwise, \code{name} is \emph{sanitized}. If the sanitized string is no
+#'   greater than \code{maxSize} characters long and is not empty, it is
+#'   returned. Otherwise, \code{defaultName} is returned. The sanitization
+#'   process replaces sequences of two or more '.' characters with a single '.'
+#'   and removes any non-alphanumeric characters.
```

Comment:
For non-exported functions, you should use either `@keywords internal` so that the function doesn't show up in the package's help index, or use regular comments (lines starting with `#` instead of `#'`) so that no documentation file is generated. We usually do the latter.